src/main/webapp/app/overview/course-score-calculation.service.ts

```diff
@@ -26,20 +28,39 @@ export class CourseScoreCalculationService {
     calculateTotalScores(courseExercises: Exercise[], course: Course): Map<string, number> {
         const scores = new Map<string, number>();
         let pointsAchievedByStudentInCourse = 0.0;
-        let maxPointsInCourse = 0;
-        let reachableMaxPointsInCourse = 0;
+        let maxPointsInCourse = 0; // the sum of all included exercises, whose due date is over or unset or who are automatically assessed and assessment is done
+        let reachableMaxPointsInCourse = 0; // the sum of all included and already assessed exercises
         let presentationScore = 0;
         for (const exercise of courseExercises) {
             const isExerciseFinished = !exercise.dueDate || exercise.dueDate.isBefore(dayjs());
             const isAssessmentOver = !exercise.assessmentDueDate || exercise.assessmentDueDate.isBefore(dayjs());
             const isExerciseIncluded = exercise.includedInOverallScore !== IncludedInOverallScore.NOT_INCLUDED;
-
-            if (isExerciseFinished && isExerciseIncluded) {
+            const isExerciseAssessedAutomatically = exercise.type === ExerciseType.PROGRAMMING && exercise.assessmentType === AssessmentType.AUTOMATIC;
+            const isAutomaticAssessmentFinished =
+                isExerciseAssessedAutomatically &&
+                (!(exercise as ProgrammingExercise).buildAndTestStudentSubmissionsAfterDueDate ||
+                    (exercise as ProgrammingExercise).buildAndTestStudentSubmissionsAfterDueDate!.isBefore(dayjs()));
```

Comment:
This could be extracted into a small helper method.

Extracting all the conditions and replacing the condition of the `if` in line 52 by a single method call would be even nicer, but I suppose that is not easily possible as you need to access `isAssessmentOver || isAutomaticAssessmentFinished` inside the `if` a second time.

Reply:
I extracted it into several helper methods in d2812ba